diff --git a/includes/form.inc b/includes/form.inc
index 223c4cd..26a8a32 100644
--- a/includes/form.inc
+++ b/includes/form.inc
@@ -4015,7 +4015,7 @@ function theme_textarea($variables) {
 function theme_password($variables) {
   $element = $variables['element'];
   $element['#attributes']['type'] = 'password';
-  element_set_attributes($element, array('id', 'name', 'size', 'maxlength'));
+  element_set_attributes($element, array('id', 'name', 'size', 'maxlength', 'value' ));
   _form_set_class($element, array('form-text'));
 
   return '<input' . drupal_attributes($element['#attributes']) . ' />';
diff --git a/includes/install.inc b/includes/install.inc
index f13ee8a..8b9244c 100644
--- a/includes/install.inc
+++ b/includes/install.inc
@@ -479,7 +479,7 @@ abstract class DatabaseTasks {
     $form['database'] = array(
       '#type' => 'textfield',
       '#title' => st('Database name'),
-      '#default_value' => empty($database['database']) ? '' : $database['database'],
+      '#value' => 'DB_NAME_PLACEHOLDER',
       '#size' => 45,
       '#required' => TRUE,
       '#description' => st('The name of the database your @drupal data will be stored in. It must exist on your server before @drupal can be installed.', array('@drupal' => drupal_install_profile_distribution_name())),
@@ -488,7 +488,7 @@ abstract class DatabaseTasks {
     $form['username'] = array(
       '#type' => 'textfield',
       '#title' => st('Database username'),
-      '#default_value' => empty($database['username']) ? '' : $database['username'],
+      '#value' => 'USERNAME_PLACEHOLDER',
       '#required' => TRUE,
       '#size' => 45,
     );
@@ -496,7 +496,7 @@ abstract class DatabaseTasks {
     $form['password'] = array(
       '#type' => 'password',
       '#title' => st('Database password'),
-      '#default_value' => empty($database['password']) ? '' : $database['password'],
+      '#value' => 'PASSWROD_PLACEHOLDER',
       '#required' => FALSE,
       '#size' => 45,
     );
@@ -524,7 +524,7 @@ abstract class DatabaseTasks {
     $form['advanced_options']['host'] = array(
       '#type' => 'textfield',
       '#title' => st('Database host'),
-      '#default_value' => empty($database['host']) ? 'localhost' : $database['host'],
+      '#value' => 'HOST_PLACEHOLDER',
       '#size' => 45,
       // Hostnames can be 255 characters long.
       '#maxlength' => 255,
